Code/app/src/main/java/ca/ualberta/moodroid/ui/AddMoodDetail.java

```diff
@@ -123,45 +213,83 @@ public void onTimeSet(TimePicker timePicker, int hour, int minute) {
                 updateTimeDisplay();
             }
         };
+
+        addPhotoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent();
+                intent.setType("image/*");
+                intent.setAction(Intent.ACTION_GET_CONTENT);
+                startActivityForResult(Intent.createChooser(intent, "choose photo"), 1);
+            }
+        });
+
+
     }
 
+    /**
+     * Update date display.
+     */
     public void updateDateDisplay() {
         this.date.setText(this.getDateString());
     }
 
+    /**
+     * Gets date string.
+     *
+     * @return the date string
+     */
     public String getDateString() {
         return new SimpleDateFormat("MM/dd/yy", Locale.US).format(calendar.getTime());
     }
 
+    /**
+     * Update time display.
+     */
     public void updateTimeDisplay() {
         this.time.setText(this.getTimeString());
     }
 
+    /**
+     * Gets time string.
+     *
+     * @return the time string
+     */
     public String getTimeString() {
         return new SimpleDateFormat("HH:mm", Locale.US).format(calendar.getTime());
     }
 
+    /**
+     * Show the date picker dialog
+     */
     @OnClick(R.id.mood_detail_date)
     public void dateClick() {
         Log.d("MOODDETAIL/DATE", "Date clicked!");
         new DatePickerDialog(AddMoodDetail.this, dateDialog, calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH)).show();
 
     }
 
+    /**
+     * Show the time picker dialog
+     */
     @OnClick(R.id.mood_detail_time)
     public void timeClick() {
         Log.d("MOODDETAIL/DATE", "Time clicked!");
         new TimePickerDialog(AddMoodDetail.this, timeDialog, calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), true).show();
     }
 
-
+    /**
+     * On click of the confirm button, create the new mood event and direct the user to the mood history view
+     */
     @OnClick(R.id.add_detail_confirm_btn)
     public void confirmClick() {
+        uploadPhoto();      /*upload photo to firebase storage */
```

Comment:
This probably needs to be an asynchronous task , that once completed will let is create the mood event.